addons/cookoff/functions/fnc_cookOff.sqf

```diff
@@ -30,6 +30,8 @@ TRACE_9("cooking off",_vehicle,_intensity,_instigator,_smokeDelayEnabled,_ammoDe
 if (_vehicle getVariable [QGVAR(isCookingOff), false]) exitWith {};
 _vehicle setVariable [QGVAR(isCookingOff), true, true];
 
+[QGVAR(addCleanupHandlers), [_vehicle]] call CBA_fnc_localEvent;
```

Comment:
1. zeus spawns vehicle
2. vehicle cookoffs
3. zeus disconnects
4. something deletes vehicle
5. effects stay there?

shouldn't this be global?

If so the locality check in deleted EH could be dropped and event should be local then.

Reply:
This is a good point, but isn't the simplest way to do this to add the EH on the server as a server event? This way no matter where the vehicle ends up local at time of deletion, the global cleanup event can be fired from a single location which will always exists for the duration of the session.

Reply:
To clarify that: The deleted EH needs to run once and send out a global event for cleanup to handle JIPs where it may not have been first added (unless I'm misremembering my Arma 3 locality logic).

Reply:
True, server event would be the best. Add cleanup handler on server, emit global event from server.

Reply:
I'll take a look at reworking this this weekend, totally slipped my mind it'd need to handle locality changing hands.